bson/bsonrw/value_reader.go

```diff
@@ -748,7 +748,11 @@ func (vr *valueReader) readBytes(length int32) ([]byte, error) {
 
 	start := vr.offset
 	vr.offset += int64(length)
-	return vr.d[start : start+int64(length)], nil
+
+	b := make([]byte, length)
+	copy(b, vr.d[start:start+int64(length)])
```

Comment:
Depending on the result of #891, this copy may be moved to the `ReadBinary` function (cc @matthewdale), in which case we should probably do the same on the `release/1.7` branch.

Reply:
I just opened https://github.com/mongodb/mongo-go-driver/pull/891 for review. Agreed we should review that before back-porting this fix.

Reply:
Thank you for the heads up. I have added the commit onto this PR.